source/opt/instruction.h

```diff
@@ -576,9 +581,6 @@ inline void Instruction::ForEachInst(
 inline void Instruction::ForEachId(const std::function<void(uint32_t*)>& f) {
   for (auto& opnd : operands_)
     if (spvIsIdType(opnd.type)) f(&opnd.words[0]);
-  if (type_id_ != 0u) type_id_ = GetSingleWordOperand(0u);
```

Comment:
Wow. that was pretty weird.

Reply:
Yea, those are the lines which triggered this in the first place, heh.